modules/utxo-core/src/bip322/toSign.ts

```diff
@@ -0,0 +1,49 @@
+import { Psbt, Transaction } from '@bitgo/utxo-lib';
+
+export type AddressDetails = {
+  redeemScript?: Buffer;
+  witnessScript?: Buffer;
+};
+
+/**
+ * Construct the toSign PSBT for a BIP322 verification.
+ * Source implementation:
+ * https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki#full
+ *
+ * @param {string} toSpendTxHex - The hex representation of the `toSpend` transaction.
+ * @param {AddressDetails} addressDetails - The details of the address, including redeemScript and/or witnessScript.
+ * @returns {string} - The hex representation of the constructed PSBT.
+ */
+export function buildToSignPsbt(toSpendTxHex: string, addressDetails: AddressDetails): string {
```

Comment:
stringly typed:

pass a higher level type like `Transaction` instead and return a higher level type like `Psbt`